### PR TITLE
SepaDelegate: Add delegate for sepa component

### PR DESCRIPTION
**Pull request by ozgur00**

### CODE REVIEW

File: sepa/src/main/java/com/adyen/checkout/sepa/DefaultSepaDelegate.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+import com.adyen.checkout.components.model.payments.request.PaymentComponentData
+import com.adyen.checkout.components.model.payments.request.SepaPaymentMethod
+import com.adyen.checkout.components.util.PaymentMethodTypes
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+
+class DefaultSepaDelegate(private val paymentMethod: PaymentMethod) : SepaDelegate {
```
Comment (OscarSpruit) [inline sepa/src/main/java/com/adyen/checkout/sepa/DefaultSepaDelegate.kt:13]: This class can be internal

File: sepa/src/main/java/com/adyen/checkout/sepa/DefaultSepaDelegate.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    override val outputDataFlow: Flow<SepaOutputData?> = _outputDataFlow
+
+    private val _componentStateFlow = MutableStateFlow<PaymentComponentState<SepaPaymentMethod>?>(null)
+    override val componentStateFlow: Flow<PaymentComponentState<SepaPaymentMethod>?> = _componentStateFlow
+
+    override fun getPaymentMethodType(): String {
+        return paymentMethod.type ?: PaymentMethodTypes.UNKNOWN
+    }
+
+    override fun onInputDataChanged(inputData: SepaInputData) {
```
Comment (OscarSpruit) [inline sepa/src/main/java/com/adyen/checkout/sepa/DefaultSepaDelegate.kt:25]: Let's take the opportunity to add some more tests. Could you write some for this class? You can take my GooglePayDelegate PR as example

**Review by OscarSpruit — APPROVED** [review]

File: sepa/src/main/java/com/adyen/checkout/sepa/DefaultSepaDelegate.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        createComponentState(outputData)
+    }
+
+    private fun outputDataChanged(outputData: SepaOutputData) {
+        _outputDataFlow.tryEmit(outputData)
+    }
+
+    override fun createComponentState(outputData: SepaOutputData) {
+        val paymentComponentData = PaymentComponentData<SepaPaymentMethod>()
+        val paymentMethod = SepaPaymentMethod().apply {
```
Comment (jreij) [inline sepa/src/main/java/com/adyen/checkout/sepa/DefaultSepaDelegate.kt:38]: You can use the constructor directly i think
```kotlin
SepaPaymentMethod(
            type = SepaPaymentMethod.PAYMENT_METHOD_TYPE,
            ownerName = outputData.ownerNameField.value,
            iban = outputData.ibanNumberField.value,
)
```

File: sepa/src/main/java/com/adyen/checkout/sepa/DefaultSepaDelegate.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            type = SepaPaymentMethod.PAYMENT_METHOD_TYPE
+            ownerName = outputData.ownerNameField.value
+            iban = outputData.ibanNumberField.value
+        }
+        paymentComponentData.paymentMethod = paymentMethod
```
Comment (jreij) [inline sepa/src/main/java/com/adyen/checkout/sepa/DefaultSepaDelegate.kt:43]: Here also you can just do val paymentComponentData = `PaymentComponentData<SepaPaymentMethod>(paymentMethod)`

**Review by jreij — APPROVED** [review]

**Pull request merged**